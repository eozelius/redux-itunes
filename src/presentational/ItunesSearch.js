import React, { Component } from 'react'
import style from '../styles/ItunesSearch.module.sass'
import indexStyle from '../styles/index.module.sass'

class ItunesSearch extends Component {
  constructor(props){
    super(props)
    this.state = {
      search: ''
    }
  }

  handleChange = (e) => {
    e.preventDefault()
    this.setState({
      search: e.target.value
    })
  }

  handleSubmit = (e) => {
    e.preventDefault()
    this.props.handleSearch(this.state.search)
  }

  render(){
    return(
      <div className={style.itunesSearch} >
        <form className='control-form' onSubmit={this.handleSubmit}>
          <div className='control-group'>
            <label>Search Itunes</label>
            <input type='text' onChange={this.handleChange} autoFocus />
            <input className={indexStyle.btn}  type='submit' />
          </div>
        </form>
      </div>
    )
  }
}

export default ItunesSearch
