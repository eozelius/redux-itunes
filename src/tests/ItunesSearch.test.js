import ItunesSearch from '../presentational/ItunesSearch'
import React from 'react'

describe(ItunesSearch, () => {
  test('it renders properly', () => {
    const search = shallow(<ItunesSearch />)
    expect(search).toMatchSnapshot()
  })

  test('it maintains invokes a callback when the input changes', () => {
    const search = shallow(<ItunesSearch />)
    search.find('input[type="text"]').simulate('change', {
        preventDefault: () => {},
        target: {
          value: 'asdf'
        }
      }
    )
    expect(search.state('search')).toEqual('asdf')
  })

  test('it invokes props.handleSearch when the form submits', () => {
    const mockFn = jest.fn()
    const search = shallow(<ItunesSearch handleSearch={mockFn} />)
    search.find('form').simulate('submit', {preventDefault: () => {}})
    expect(mockFn).toHaveBeenCalledWith('')
  })
})
