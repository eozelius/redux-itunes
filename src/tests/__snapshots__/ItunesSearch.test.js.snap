// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItunesSearch it renders properly 1`] = `
<div
  className="itunesSearch"
>
  <form
    className="control-form"
    onSubmit={[Function]}
  >
    <div
      className="control-group"
    >
      <label>
        Search Itunes
      </label>
      <input
        autoFocus={true}
        onChange={[Function]}
        type="text"
      />
      <input
        className="btn"
        type="submit"
      />
    </div>
  </form>
</div>
`;
